zb-db/src/main/java/io/zeebe/db/impl/rocksdb/transaction/ZeebeTransactionDb.java

```diff
@@ -143,49 +128,56 @@ protected long getColumnFamilyHandle(ColumnFamilyNames columnFamily) {
   public <KeyType extends DbKey, ValueType extends DbValue>
       ColumnFamily<KeyType, ValueType> createColumnFamily(
           ColumnFamilyNames columnFamily, KeyType keyInstance, ValueType valueInstance) {
-    return new RocksDbColumnFamily<>(this, columnFamily, keyInstance, valueInstance);
+    return new TransactionalColumnFamily<>(this, columnFamily, keyInstance, valueInstance);
   }
 
   protected void put(long columnFamilyHandle, DbKey key, DbValue value) {
-    key.write(keyBuffer, 0);
-    value.write(valueBuffer, 0);
+    ensureInOpenTransaction(
+        () -> {
+          key.write(keyBuffer, 0);
+          value.write(valueBuffer, 0);
 
-    try {
-      if (isInBatch()) {
-        batch.put(columnFamilyHandle, key, value);
-      } else {
-        put(
-            nativeHandle_,
-            keyBuffer.byteArray(),
-            0,
-            key.getLength(),
-            valueBuffer.byteArray(),
-            0,
-            value.getLength(),
-            columnFamilyHandle);
-      }
-    } catch (Exception e) {
-      throw new RuntimeException("Unexpected error occurred trying to write RocksDB entry", e);
-    }
+          try {
+            currentTransaction.put(
+                columnFamilyHandle,
+                keyBuffer.byteArray(),
+                key.getLength(),
+                valueBuffer.byteArray(),
+                value.getLength());
+          } catch (Exception e) {
```

Comment:
Can we remove all those catch statements and have a single one in `transaction`? We're anyway catching everything there.